treq/auth.py

```diff
@@ -26,6 +27,31 @@ def request(self, method, uri, headers=None, bodyProducer=None):
             method, uri, headers=headers, bodyProducer=bodyProducer)
 
 
+class _PinToFirstHostAgent(object):
+    """
+    An {twisted.web.iweb.IAgent} implementing object that takes two
+    agents, using the first as when the current request's host name
+    matches first request's host name, and the second when it does
+    not.
+    """
+
+    def __init__(self, first_agent, second_agent):
+        self._first_agent = first_agent
```

Comment:
Lots of [treq.client](https://github.com/twisted/treq/blob/master/src/treq/client.py), for example, prefers `underscore_names`.  I'd rather confront naming consistency in a subsequent PR and stick with pep8 names in this one, since the API I'm extending [prefers underscores](https://github.com/twisted/treq/blob/master/src/treq/auth.py#L37)